### PR TITLE
FINERACT-835: Upgrade Swagger and automatically generate the JSON spec

### PR DESCRIPTION
**Pull request by ptuomola**
FINERACT-835

## Description
This PR includes:

- Upgrading Swagger UI to the latest version (3.26.0)
- Making Swagger UI a packaged dependency rather than checking in the individual files
- Exposing Swagger UI in /fineract-provider/swagger-ui with response.json as the input
- Adding Swagger Plugin to autogenerate the JSON spec
- Configuring Gradle tasks to build the JSON spec automatically after build

The Swagger UI does not yet point to the new JSON (fineract.json) but instead points to the previous manually created response.json, which remains checked in. The next step would be to ensure the autogenerated fineract.json contains all the same information as the response.json. After that, the response.json can be deleted. 

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [x] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by vorburger** [comment]
Whoa! @ptuomola FYI @thesmallstar was going to look into Swagger after completing Checkstyle (and SQL security), but I'm sure he doesn't mind this start.. we have, more than, enough to do on this project for everyone! 

I'll have a closer look at this myself also some time in the coming days.

**Comment by ptuomola** [comment]
@vorburger @thesmallstar Not to worry - there's still plenty to do on Swagger! I just did the basics: upgrading & fixed Swagger UI and added the Gradle task to create the JSON. 

All the rest of the tasks still remain, i.e.
- Getting the rest of the information into the generated JSON file
- Improving layout / navigation of Swagger
- Getting "Try me" button to work
- Autogenerating client library
etc etc...

**Comment by thesmallstar** [comment]
@ptuomola I really appreciate this work, I want to wrap SQL security and Swagger quickly (and nicely) so I can look into more functional issues at Fineract, and this would really help :D

File: README.md [hunk 1]
```diff
@@ -251,6 +251,8 @@ Apache Fineract Platform API
 
 The API for the Fineract-platform (project named 'Apache Fineract') is documented in the API-docs under <b><i>Full API Matrix</i></b> and can be viewed [here](https://demo.fineract.dev/fineract-provider/api-docs/apiLive.htm "API Documentation").
 
+If you have your own Fineract instance running, you can find the same documentation under '(your host:port)/fineract-provider/api-docs/apiLive.htm' under your. The Swagger documentation (work in progress) can be accessed under '(your host:port)/fineract-provider/swagger-ui/index.html'
```
Comment (vorburger) [inline README.md:254]: Very very minor, but "under your" is duplicated, better like this:

```suggestion
If you have your own Fineract instance running, you can find the same documentation under '(your host:port)/fineract-provider/api-docs/apiLive.htm'. The Swagger documentation (work in progress) can be accessed under '(your host:port)/fineract-provider/swagger-ui/index.html'
```

File: fineract-provider/src/main/resources/static/swagger-ui/index.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<!-- HTML for static distribution bundle build -->
```
Comment (vorburger) [inline fineract-provider/src/main/resources/static/swagger-ui/index.html:1]: @ptuomola just out of curiosity and for my education (and FYI @thesmallstar), does `org.webjars.npm:swagger-ui-dist` not already include an `index.html` like this? If we can avoid having to maintain this, that's better, of course. If in Swagger (which I don't know full details about) an application needs to provide this itself, then fine, of course - just thought I would ask.

**Review by vorburger — APPROVED** [review]
I'll merge this now, as a first step towards working Swagger in Fineract, but it's important to note that this is a first step - we do NOT really have working Swagger, yet - but incremental small steps are the road to success! Details provided in FINERACT-835.

File: fineract-provider/src/main/resources/static/swagger-ui/index.html (hunk 2 above)
Comment (ptuomola) [reply fineract-provider/src/main/resources/static/swagger-ui/index.html:1]: @vorburger The swagger-ui-dist does include an index.html, but that one points to petstore's JSON spec by default. The Swagger UI Docker image etc allows overriding the default JSON spec location with variables, but the WebJar package doesn't seem to. Hence I needed to override the default /swagger-ui/index.html to allow us to control the JSON spec that gets loaded by default (i.e. currently response.json, in the future fineract.json)

**Comment by Sanyam96** [comment]
Thanks, @ptuomola for this PR.
This PR is a progress on swagger on Fineract. 
@vorburger  thanks for reviewing this PR and merging it so quickly.

**Pull request merged**